amd/src/pxaiendpoint.js

```diff
@@ -1,171 +1,195 @@
+ */
+export const init = (
+    id,
+    cmid,
+    contextid,
+    steps_data,
+    assignmentid,
+    attempt_text
+) => {
+
+    let EventCreator = function (stepsData, assignmentId, attemptText) {
         this.currStep = 0;
         this.selStart = 0;
         this.selEnd = 0;
-        this.assignmentid = assignmentid
-        this.attempt_text = attempt_text
-
+        this.assignmentid = assignmentId;
+        this.attempt_text = attemptText;
         this.init();
-
-    }
+    };
 
     EventCreator.prototype.init = function () {
 
         this.setAttemtCountText(this.attempt_text);
+        const container = $(".actions-container");
+
+        /**
+         * @param {HTMLElement} button
+         */
+        const highlightStepButton = (button) => {
+            if (!(button instanceof HTMLElement)) {
+                return;
+            }
+            button.classList.add('btn-outline-primary');
+            button.classList.remove('btn-outline-secondary');
+        };
+
+        /**
+         * @param {HTMLElement} button
+         */
+        const blurStepButton = (button) => {
+            if (!(button instanceof HTMLElement)) {
+                return;
+            }
+            button.classList.remove('btn-outline-primary');
+            button.classList.add('btn-outline-secondary');
+        };
 
-        var getSelectedTextareaText = function (name) {
+        let getSelectedTextareaText = function (name) {
 
-            var selText = '';
-            var txtarea = document.getElementsByName(name);
-            txtarea = (txtarea && txtarea.length > 0) ? txtarea[0] : null;
-            if (txtarea) {
-                var start = txtarea.selectionStart;
+            let selText = '';
+            let textArea = document.getElementsByName(name);
+            textArea = (textArea && textArea.length > 0) ? textArea[0] : null;
+            if (textArea) {
+                let start = textArea.selectionStart;
                 this.selStart = start;
-                var finish = txtarea.selectionEnd;
+                let finish = textArea.selectionEnd;
                 this.selEnd = finish;
-                selText = txtarea.value.substring(start, finish);
+                selText = textArea.value.substring(start, finish);
             } else {
-                console.log("pxaiwriter : no known text selecable source");
+                window.console.log("pxaiwriter : no known text selecable source");
             }
             return selText;
 
         }.bind(this);
 
-        $(".actions-container").on("click", '#pxaiwriter-expand-selection', function (e) {
+        document.querySelector('.assignsubmission_pxaiwriter').addEventListener(AISubmissionEvent.EventList.pageChange, (e) => {
+
+            const event = new AISubmissionPageChangeEvent(e);
+            const currentStepSelector = `.step-page-button[data-step-number="${event.currentStep}"]`;
+            const allStepSelector = `.step-page-button[data-step-number]`;
```

Comment:
Seems a bit arbitrary to have const for currentStepSelector and allStepSelector, when they are only used once. Why not just have the selectors directly in the querySelector?

Reply:
It's easier to understand what kind of selector is it, instead of just string.